orchagent/portsorch.cpp

```diff
@@ -5757,6 +5935,33 @@ void PortsOrch::updatePortOperStatus(Port &port, sai_port_oper_status_t status)
     if (port.m_type == Port::PHY)
     {
         updateDbPortOperStatus(port, status);
+        if (port.m_admin_state_up && port.m_autoneg > 0)
+        {
+            if (status == SAI_PORT_OPER_STATUS_UP)
+            {
+                updatePortStateAutoNeg(port);
+                updatePortStatePoll(port, PORT_STATE_POLL_AN, false);
+            }
+            else
+            {
+                /* Restart autoneg state polling upon link down event */
+                m_portStateTable.hdel(port.m_alias, "rmt_adv_speeds");
+                updatePortStatePoll(port, PORT_STATE_POLL_AN, true);
+            }
+        }
+        if (port.m_admin_state_up && port.m_link_training > 0)
+        {
+            updatePortStateLinkTraining(port);
```

Comment:
Should this move inside the below if condition?

Reply:
We already have the if-statement for the port admin state and LT enablement, may I have more details of the additional checker that's missing? Thank you.

Reply:
In the case of AN
- Link Up: Refresh AN states
- Link Down: Clear the remote advertisement
In the case of LT
- Link Up: Refresh LT states
- Link Down: Refresh LT states

Hence the updatePortStateLinkTraining() is invoked before checking the operation link status